Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -188,14 +188,60 @@ internal Status GetStatus()
             {
                 var grpcStatus = OriginalResponseMessage.StatusCode switch
                 {
-                    HttpStatusCode.OK => StatusCode.OK,
-                    HttpStatusCode.Unauthorized => StatusCode.Unauthenticated,
-                    HttpStatusCode.Forbidden => StatusCode.PermissionDenied,
+                    // Note: this switch statement contains all the unique named values within HttpStatusCode (as of 2021-08-09).
+
+                    HttpStatusCode.Accepted => StatusCode.OK,
+                    HttpStatusCode.Ambiguous => StatusCode.Unknown,
+                    HttpStatusCode.BadGateway => StatusCode.InvalidArgument,
```

Comment:
This doesn't seem consistent with https://grpc.github.io/grpc/core/md_doc_http-grpc-status-mapping.html

Reply:
That table seems to be losing a *lot* of information though, and isn't really aimed at our use case, IMO. Mapping almost everything to unknown seems like a bad idea to me.

I'm happy to revisit specific ones like BadGateway (and Unavailable seems fine for that) but I don't think we should follow that table for everything. Where there's a really clear mapping (e.g. BadRequest to InvalidArgument) I don't see why we'd want to lose that information.